src/earn/EarnEnterAmount.tsx

```diff
@@ -202,30 +234,39 @@ function EarnEnterAmount({ route }: Props) {
         localAmount: parsedLocalAmount,
       }
     }
-  }, [tokenAmountInput, localAmountInput, enteredIn, token])
+  }, [tokenAmountInput, localAmountInput, enteredIn, transactionToken, mode])
 
-  const feeCurrencies = useSelector((state) => feeCurrenciesSelector(state, token.networkId))
+  const feeCurrencies = useSelector((state) =>
+    feeCurrenciesSelector(state, transactionToken.networkId)
+  )
 
   useEffect(() => {
     clearPreparedTransactions()
 
     if (
       !tokenAmount ||
       tokenAmount.isLessThanOrEqualTo(0) ||
-      tokenAmount.isGreaterThan(token.balance)
+      tokenAmount.isLessThanOrEqualTo(0) ||
+      tokenAmount.isGreaterThan(
+        isWithdrawal
+          ? transactionToken.balance.multipliedBy(pool.pricePerShare[0])
+          : transactionToken.balance
+      )
     ) {
       return
     }
     const debouncedRefreshTransactions = setTimeout(() => {
-      return handleRefreshPreparedTransactions(tokenAmount, token, feeCurrencies)
+      return handleRefreshPreparedTransactions(tokenAmount, transactionToken, feeCurrencies)
     }, FETCH_UPDATED_TRANSACTIONS_DEBOUNCE_TIME)
     return () => clearTimeout(debouncedRefreshTransactions)
-  }, [tokenAmount, token])
+  }, [tokenAmount, mode, transactionToken, feeCurrencies])
 
   const { estimatedFeeAmount, feeCurrency, maxFeeAmount } =
     getFeeCurrencyAndAmounts(prepareTransactionsResult)
 
-  const isAmountLessThanBalance = tokenAmount && tokenAmount.lte(token.balance)
+  const isAmountLessThanBalance = isWithdrawal
+    ? tokenAmount && tokenAmount.lte(transactionToken.balance.multipliedBy(pool.pricePerShare[0]))
```

Comment:
seems like this calculation is done in multiple places. would be nice to have `amountInTransactionToken` (or something similar) and use that everywhere, vs doing different comparisons for `withdraw` and `deposit`

Reply:
Added in 686a01c922903dd3e1736f5b9a1ced0b40a44c76!